pkgs/development/libraries/mozc/default.nix

```diff
@@ -0,0 +1,241 @@
+, glibcLocales
+, lib
+, pkg-config
+, python3
+, qtbase
+, qtwayland
+, ruby
+, wrapQtAppsHook
+, dictionaries ? [
+    "alt-cannadic"
+    "edict2"
+    "jawiki"
+    "neologd"
+    "personal-names"
+    "place-names"
+    "skk-jisyo"
+    "sudachidict"
+  ]
+}:
+
+buildBazelPackage {
+  pname = "mozc";
+  version = "unstable-2024-02-09";
+
+  srcs = [
+    (fetchFromGitHub rec {
+      owner = "google";
+      repo = "mozc";
+      name = repo;
+      fetchSubmodules = true;
+      rev = "c2fcbf6515c5884437977de46187c16a8cb7bb50";
+      hash = "sha256-AcIN5sWPBe4JotAUYv1fytgQw+mJzdFhKuVPLR48soA=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "merge-ut-dictionaries";
+      name = repo;
+      rev = "a3d6fc4005aff2092657ebca98b9de226e1c617f";
+      hash = "sha256-UK29ACZUK9zGfzW7C85uMw2aF5Gk+0aDeUdNV71PY+0=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-alt-cannadic";
+      name = repo;
+      rev = "4e548e6356b874c76e8db438bf4d8a0b452f2435";
+      hash = "sha256-4gzqVoCIhC0k3mh0qbEr8yYttz9YR0fItkFNlu7cYOY=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-edict2";
+      name = repo;
+      rev = "b2eec665b81214082d61acee1c5a1b5b115baf1a";
+      hash = "sha256-LIpGt6xB8dLUnazbJHZk6EH1/ZyAHMIn1m6Qpr2dsHs=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-jawiki";
+      name = repo;
+      rev = "6e08b8c823f3d2d09064ad2080e7a16552a7b473";
+      hash = "sha256-0YwAinlcI6yojCdW1MpLgMZfyYV7gk9Q+Wlu4lR3Hrg=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-neologd";
+      name = repo;
+      rev = "bf9d0d217107f2fb2e7d1a26648ef429d9fdcd27";
+      hash = "sha256-e0iM5fohwpNNhPl9CjkD753/Rgatg7GdwN0NSvlN94c=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-personal-names";
+      name = repo;
+      rev = "8a500f82c553936cbdd33b85955120e731069d44";
+      hash = "sha256-pMyYvl5S0+U++MO5m9rmbtxDzAmO4Xs8sFewOUGqgUA=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-place-names";
+      name = repo;
+      rev = "a847a02e0137ab9e2fdbbaaf120826f870408ca6";
+      hash = "sha256-B0kW8Wa/nCT4KEYl2Rz6gQcj0Po3GxU6i42unHhgZeU=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-skk-jisyo";
+      name = repo;
+      rev = "ee94f6546ce52edfeec0fd203030f52d4d99656f";
+      hash = "sha256-RXxO878ZBkxenrdo7cFom5NjM0m7CdYQk0dFu/HPp/Y=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-sudachidict";
+      name = repo;
+      rev = "55f61c3fca81dec661c36c73eb29b2631c8ed618";
+      hash = "sha256-gNnBcuVU1M7rllfZXIrLg7WYUhKqPJsUjR8Scnq3Fw8=";
+    })
+    (fetchurl rec {
+      name = "jawiki";
+      url = "https://dumps.wikimedia.org/${name}/20240120/${name}-20240120-all-titles-in-ns0.gz";
```

Comment:
This link is dead. I think we'll need to revendor it to keep it alive, waybackmachine does not seem to like it. As far as I understand, revendoring should be fine under GFDL, see https://dumps.wikimedia.org/legal.html

Reply:
I also tried it with waybackmachine and it worked just fine
https://web.archive.org/web/20240822091528/https://dumps.wikimedia.org/jawiki/20240801/jawiki-20240801-all-titles-in-ns0.gz

Reply:
I made a repository that automatically archives it, so that that this package can lock on a specific revision: https://github.com/musjj/jawiki-archive